src/util/uploadExternalFormatFile.tsx

```diff
@@ -0,0 +1,150 @@
+export interface UploadExternalFormatFileFormValues {
+  imageFile: File | null;
+  name: string;
+  pool: string;
+  formatConversion: boolean;
+  virtioConversion: boolean;
+  architecture: string;
+}
+
+export const FILE_CHUNK_SIZE = 1024 * 1024;
+
+export const supportedVMArchOptions = (hostArchitectures: string[]) => {
```

Comment:
```suggestion
export const externalUploadArchitectureOptions = (hostArchitectures: string[]) => {
```
Maybe clearer?

Reply:
hmm not sure about this one, since these values are VM architectures supported by LXD in general and does not only relate to VM image uploads.

Reply:
I see, then the existing name seems correct.